dafni_cli/api/session.py

```diff
@@ -239,11 +247,40 @@ def login(username: str, password: str):
             )
         return DAFNISession(SessionData.from_login_response(username, login_response))
 
-    def _request_user_login(self):
+    def _attempt_login_from_env(self) -> bool:
+        """Attempts to login using environment variables (if found)
+
+        If environment variables are found but login fails will cause
+        the program to exit.
+
+        Returns:
+            bool: True if a username and password were found in the
+                  environment, False otherwise
         """
-        Prompts the user for their username and password. If login is successful,
-        notifies the user that login has been completed and displays the username
-        and UUID.
+
+        username = os.getenv("DAFNI_USERNAME")
+        password = os.getenv("DAFNI_PASSWORD")
+
+        if username is not None and password is not None:
+            # Attempt login
+            login_response = self._login(username, password)
+
+            if not login_response.was_successful():
```

Comment:
I wonder if instead of exiting if this fails it should give the command line prompt for username and password? Or are we thinking the main use case for this would be in scripting and it would be better to error out?

Reply:
Yeah I was specifically thinking of if it was used in a script somewhere - prompting for a username/password in this case could make it hang indefinitely so I thought it was better to just exit with an error. If used in github actions for example this should cause it to fail.